pkg/controller/siddhiprocess/siddhiprocess_controller.go

```diff
@@ -105,171 +105,160 @@ type ReconcileSiddhiProcess struct {
-			reqLogger.Info("Ingress created successfully")
-			return reconcile.Result{Requeue: true}, nil
-		} else if err == nil{
-			siddhiIngress := reconcileSiddhiProcess.updatedLoadBalancerForSiddhiProcess(siddhiProcess, ingress, siddhiApp)
-			reqLogger.Info("Updating a new Ingress", "Ingress.Namespace", siddhiIngress.Namespace, "Ingress.Name", siddhiIngress.Name)
-			err = reconcileSiddhiProcess.client.Update(context.TODO(), siddhiIngress)
+			sp.Status.Status = getStatus(RUNNING)
+			err = rsp.client.Status().Update(context.TODO(), sp)
 			if err != nil {
-				reqLogger.Error(err, "Failed to updated new Ingress", "Ingress.Namespace", siddhiIngress.Namespace, "Ingress.Name", siddhiIngress.Name)
-				return reconcile.Result{}, err
+				reqLogger.Error(err, "Failed to update SiddhiProcess status")
+				sp.Status.Status = getStatus(ERROR)
 			}
-			// Ingress updated successfully - return and requeue
-			reqLogger.Info("Ingress updated successfully")
 			return reconcile.Result{Requeue: true}, nil
-		} else if err != nil{
-			reqLogger.Error(err, "Failed to get Ingress")
+		} else if err != nil {
+			reqLogger.Error(err, "Failed to get Service")
+			sp.Status.Status = getStatus(ERROR)
 			return reconcile.Result{}, err
 		}
-	}
 
+		createIngress := true
+		if (operatorEnvs["AUTO_INGRESS_CREATION"] != "") && (operatorEnvs["AUTO_INGRESS_CREATION"] != "false") {
+			createIngress = true
+		} else {
+			createIngress = false
+		}
+
+		if createIngress {
+			ingress := &extensionsv1beta1.Ingress{}
+			err = rsp.client.Get(context.TODO(), types.NamespacedName{Name: "siddhi", Namespace: sp.Namespace}, ingress)
+			if err != nil && errors.IsNotFound(err) {
+				siddhiIngress := rsp.loadBalancerForSiddhiProcess(sp, siddhiApp, configs)
+				reqLogger.Info("Creating a new Ingress", "Ingress.Namespace", siddhiIngress.Namespace, "Ingress.Name", siddhiIngress.Name)
+				err = rsp.client.Create(context.TODO(), siddhiIngress)
+				if err != nil {
+					reqLogger.Error(err, "Failed to create new Ingress", "Ingress.Namespace", siddhiIngress.Namespace, "Ingress.Name", siddhiIngress.Name)
+					return reconcile.Result{}, err
+				}
+				reqLogger.Info("Ingress created successfully")
+				return reconcile.Result{Requeue: true}, nil
+			} else if err == nil {
+				siddhiIngress := rsp.updatedLoadBalancerForSiddhiProcess(sp, ingress, siddhiApp, configs)
+				reqLogger.Info("Updating a new Ingress", "Ingress.Namespace", siddhiIngress.Namespace, "Ingress.Name", siddhiIngress.Name)
+				err = rsp.client.Update(context.TODO(), siddhiIngress)
+				if err != nil {
+					reqLogger.Error(err, "Failed to updated new Ingress", "Ingress.Namespace", siddhiIngress.Namespace, "Ingress.Name", siddhiIngress.Name)
+					return reconcile.Result{}, err
+				}
+				reqLogger.Info("Ingress updated successfully")
+				return reconcile.Result{Requeue: true}, nil
+			} else if err != nil {
+				reqLogger.Error(err, "Failed to get Ingress")
```

Comment:
Shall we try to add context to the error if possible. eg: failed ingress name, port we tried to create the ingress and so on.
Please check all places